tropohelper/services.py

```diff
@@ -26,6 +36,51 @@ def create_firehose(stack, name, bucket_arn, buffering_seconds=300, buffering_si
             RoleARN="arn:aws:iam::347225174248:role/firehose_delivery_role")
         ))
 
+def create_kinesis_stream(stack, name, shard_count):
+    """Add Kinesis Stream with the specified shard count and default retention period."""
+    return stack.stack.add_resource(Stream(
+        '{0}Stream'.format(name.replace('-', '')),
+        ShardCount=shard_count
+    ))
+
+def create_json_redshift_firehose_from_stream(stack, name, firehose_arn,
+                                              source_stream_arn, source_stream_role_arn,
+                                              redshift_host_port, redshift_db_name,
+                                              redshift_username, redshift_password,
+                                              redshift_db_table_name,
+                                              s3_bucket_arn, s3_buffering_seconds=300, s3_buffering_size=5):
+    """Add Kinesus Redshift Firehose Resource with another Kinesis Stream as source."""
+    return stack.stack.add_resource(DeliveryStream(
+        '{0}Firehose'.format(name.replace('-', '')),
+        DeliveryStreamName=name,
+        DeliveryStreamType='KinesisStreamAsSource',
+        KinesisStreamSourceConfiguration=KinesisStreamSourceConfiguration(
+            KinesisStreamARN=source_stream_arn,
+            RoleARN=source_stream_role_arn
+        ),
+        RedshiftDestinationConfiguration=RedshiftDestinationConfiguration(
+            ClusterJDBCURL='jdbc:redshift://{0}/{1}'.format(redshift_host_port, redshift_db_name),
+            CopyCommand=CopyCommand(
+                CopyOptions="JSON 'auto'",
+                DataTableName=redshift_db_table_name,
+            ),
+            Password=redshift_password,
+            RoleARN=firehose_arn,
+            S3Configuration=S3Configuration(
+                BucketARN=s3_bucket_arn,
+                Prefix=name,
+                BufferingHints=BufferingHints(IntervalInSeconds=s3_buffering_seconds, SizeInMBs=s3_buffering_size),
+                CompressionFormat='GZIP',
+                EncryptionConfiguration=EncryptionConfiguration(
+                    KMSEncryptionConfig=KMSEncryptionConfig(
+                        AWSKMSKeyARN='arn:aws:kms:us-east-1:347225174248:key/232e45d1-6b15-4681-8de7-8629b0ed6b22')),
```

Comment:
We have to extract these from the code. We shouldn't have PatientPop specific stuff in here.

Reply:
extracted